provider/provider_yaml_test.go

```diff
@@ -305,6 +305,83 @@ func TestNonIdempotentSnsTopic(t *testing.T) {
 	require.ErrorContains(t, err, "already exists")
 }
 
+func TestOpenZfsFileSystemUpgrade(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without credentials")
+	}
+	const pulumiYaml = `
+name: openzfs
+runtime: yaml
+resources:
+  MyFileSystem:
+    properties:
+      deploymentType: SINGLE_AZ_1
+      storageCapacity: 64
+      %s
+      throughputCapacity: 64
+    type: aws:fsx:OpenZfsFileSystem
+  MySubnet:
+    properties:
+      cidrBlock: "10.0.1.0/24"
+      vpcId: ${MyVPC.id}
+    type: aws:ec2:Subnet
+  MyVPC:
+    properties:
+      cidrBlock: "10.0.0.0/16"
+    type: aws:ec2:Vpc
+`
+
+	var (
+		providerName    string = "aws"
+		baselineVersion string = "6.41.0"
+	)
+	cwd, err := os.Getwd()
+	assert.NoError(t, err)
+	workdir := t.TempDir()
+
+	firstProgram := []byte(fmt.Sprintf(pulumiYaml, "subnetIds: ${MySubnet.id}"))
```

Comment:
Why don't we do that in `testProviderCodeChanges`? Would be quite a bit of duplication if every test needs to do this

Reply:
which piece are you talking about? The `[]byte()` part? Everything else is specific to this test.

Reply:
Sorry, yeah the casting part. Looking again it's definitely not a lot of duplication. So rather just a nit. Feel free to ignore